targets/TARGET_NORDIC/TARGET_NRF5/TARGET_SDK11/drivers_nrf/rng/nrf_drv_rng.c

```diff
@@ -0,0 +1,250 @@
+/* Copyright (c) 2015 Nordic Semiconductor. All Rights Reserved.
+ *
+ * The information contained herein is property of Nordic Semiconductor ASA.
+ * Terms and conditions of usage are described in detail in NORDIC
+ * SEMICONDUCTOR STANDARD SOFTWARE LICENSE AGREEMENT.
+ *
+ * Licensees are granted free, non-transferable use of the information. NO
+ * WARRANTY of ANY KIND is provided. This heading must NOT be removed from
+ * the file.
+ *
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "nrf_drv_rng.h"
+#include "nrf_assert.h"
+#include "nrf_drv_common.h"
+#include "nordic_common.h"
+#include "nrf_error.h"
+#include "nrf_assert.h"
+#ifdef SOFTDEVICE_PRESENT
+#include "nrf_sdm.h"
+#include "nrf_soc.h"
+#else
+#include "app_fifo.h"
```

Comment:
Where can I find this file? It doesn't seem to be contained in the code base.

Reply:
@marcuschangarm Is it intended that this file is missing?

Reply:
Ah, I see what you are saying. `SOFTDEVICE_PRESENT` must always be defined in the current setup. Removing it will break the code in many places. The ability to remove the SoftDevice will be introduced in the SDK upgrade we are working on.

Reply:
@marcuschangarm Ok, thanks for clarifying this!

Reply:
@marcuschangarm I would prefer not to have large parts of this PR depending on a configuration which is known to be broken, though. Wouldn't it make more sense to error out in a single place if `SOFTDEVICE_PRESENT` is not defined, and adding support for `! SOFTDEVICE_PRESENT` in a single go in your SDK upgrade?

Reply:
Unfortunately, all files in the Nordic SDK have this built in bug! 😒 

We are completely re-implementing most of the mbed HAL for the new SDK and reorganizing the directory structure so that we can both change SoftDevices on the fly and even compile without them. This PR is just to get the Cloud client going until we have the new SDK in place.